src/crawlers/cheerio_crawler.js

```diff
@@ -328,21 +350,27 @@ class CheerioCrawler {
      */
     async _handleRequestFunction({ request, autoscaledPool }) {
         if (this.prepareRequestFunction) await this.prepareRequestFunction({ request });
-        const { dom, responseStream: response } = await addTimeoutToPromise(
+        const { dom, xmlMode, body, responseStream: response } = await addTimeoutToPromise(
             this._requestFunction({ request }),
             this.requestTimeoutMillis,
             'CheerioCrawler: requestFunction timed out.',
         );
 
         request.loadedUrl = response.url;
 
-        const $ = cheerio.load(dom);
+        const $ = dom ? cheerio.load(dom, { xmlMode }) : undefined;
         const context = {
             $,
             // Using a getter here not to break the original API
             // and lazy load the HTML only when needed.
             get html() {
-                return $.html({ decodeEntities: false });
+                return dom && !xmlMode ? $.html({ decodeEntities: false }) : undefined;
+            },
+            get body() {
+                if (dom) {
+                    return xmlMode ? $.xml() : $.html({ decodeEntities: false });
```

Comment:
You should just return body here

Reply:
We want to use a lazy load of HTML, only if getter is called.